pkg/cmd/bind.go

```diff
@@ -217,6 +235,36 @@ func (o *bindCmdOptions) run(cmd *cobra.Command, args []string) error {
 		binding.Spec.Integration.Traits = traits
 	}
 
+	if binding.Annotations == nil {
+		binding.Annotations = make(map[string]string)
+	}
+
+	if o.OperatorID != "" {
+		if pl, err := platformutil.LookupForPlatformName(o.Context, client, o.OperatorID); err != nil {
+			if k8serrors.IsForbidden(err) {
+				o.PrintfVerboseOutf(cmd, "Unable to verify existence of operator id [%s] due to lack of user privileges\n", o.OperatorID)
+			} else {
+				return err
+			}
+		} else if pl == nil {
+			if o.Force {
+				o.PrintfVerboseOutf(cmd, "Unable to find operator with given id [%s] - Kamelet binding may not be reconciled and get stuck in waiting state\n", o.OperatorID)
+			} else {
+				return fmt.Errorf("unable to find integration platform for given operator id '%s', use --force option or make sure to use a proper operator id", o.OperatorID)
+			}
+		}
+	}
```

Comment:
Duplicated code: we better make it an util function to be used from all the other parts

Reply:
👍